Update ovmf-descriptors for AMD SEV-SNP and Intel TDX

r1chard-lyu opened the pull request:
Update ovmf-descriptors for AMD SEV-SNP and Intel TDX
- Add 50-ovmf-x86_64-sev-snp.json descriptor for AMD SEV-SNP image
- Removed "amd-sev-snp" feature tags from 50-ovmf-x86_64-sev.json
- Updated 60-ovmf-x86_64-tdx.json descriptor

joeyli commented:
@r1chard-lyu 
Hi Richard, 
Could you please add more detail in patch description for why the descriptor must be changed?
Thanks!

r1chard-lyu committed:
Update firmware descriptors for SEV-SNP and TDX (bsc#1247847)

The firmware descriptor files for SEV-SNP and TDX currently advertise
the device type as 'pflash'. However, SNP and TDX are by design incompatible
with pflash emulation. Both do not allow the host to change guest memory
layout after launch, and pflash needs to do that to switch between reading
mode and programming mode. Instead, the device type should be identified as
'memory', allowing applications to properly map the firmware to qemu
command line option.

- Add 50-ovmf-x86_64-sev-snp.json to support the 'amd-sev-snp' feature.
- Remove the sev-snp feature from 50-ovmf-x86_64-sev.json.
- Update the device in 60-ovmf-x86_64-tdx.json from 'pflash' to 'memory'.

r1chard-lyu commented:
@joeyli 
Hi Joey,
Thank you for your reminder. 
Do you think this version of the description is improved?

joeyli commented:
It's good. Thanks a lot!
